jccguimaraes/project-viewer-plus
# (feature): add code coverage

Here you have it, @jccguimaraes . This PR adds code coverage to the package through `atom-coverage`.

To test locally just:

```shell
git clone -b new/add-code-coverage git@github.com:cgalvarez/project-viewer-plus.git
npm install
npm run test:coverage
```

It should output something like:

File|%Stmts|%Branch|%Funcs|%Lines|Uncovered Lines
---|---|---|---|---|---
All files|30.73|25.21|26.91|30.54|
--lib|41.3|10|25|41.3|
----main.js|100|100|100|100|
----package.js|41.3|10|25|41.3|...143,164,173
--lib/components|15.12|3.33|15.38|14.12|
----empty.js|53.85|25|50|53.85|...29,32,46,53
----group.js|0|0|0|0|...83,91,94,96
----icon.js|0|0|0|0|...72,80,92,96
----main.js|54.55|0|60|50|26,27,28,31,38
----project.js|0|0|0|0|...77,90,93,95
----select-list-item.js|0|0|0|0|...36,44,45,49
--lib/constants|100|100|100|100|
----base.js|100|100|100|100|
----config.js|100|100|100|100|
----icons.js|100|100|100|100|
--lib/containers|32.79|30.23|32.76|33.06|
----editor.js|0|0|0|0|...146,154,162
----icons.js|0|0|0|0|...67,68,74,82
----list.js|38.89|25|40|38.89|...60,63,64,66
----main.js|54.55|50|52.63|56.25|...177,185,192
----select-list.js|44.12|50|46.67|44.12|...136,137,145
--lib/services|40.09|38.14|38.67|39.83|
----context-switcher.js|7.81|8.33|5|7.81|...183,184,185
----database.js|39.25|21.74|35.14|39.25|...381,382,383
----devlog.js|66.67|50|100|66.67|3
----github.js|100|100|100|100|
----packages.js|100|100|100|100|
----state.js|73.58|69.57|81.25|73.08|...148,157,171
--lib/services/packages|0|0|0|0|
----find-and-replace.js|0|0|0|0|...48,50,51,54
----linter-ui-default.js|0|0|0|0|...45,47,48,51
----linter.js|0|0|0|0|...45,47,48,51
----status-bar.js|0|0|0|0|...34,37,38,40
----tree-view.js|0|0|0|0|...140,143,147

BTW, the reason why the CI builds fail is not related with this PR (at least right now), but to tests failing.
